src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -158,90 +158,65 @@ public async Task<CommitTransactionsReport> CommitTransactions(ConsistentLedgerE
-         *    commit_timestamp = {utcNow or some timestamp from committedTransactions coll},
-         *    last_failure_reason = null,
-         *    last_failure_timestamp = null,
-         *    version_control = -1
-         * FROM (
-         *    SELECT * FROM pending_transactions WHERE payload_hash = ANY(...)
-         * ) ptu
-         * WHERE pt.id = ptu.id
-         * RETURNING ptu.*;
-         *
-         * and then simple foreach loop notifying the observers
-         */
         var timestamp = _clock.UtcNow;
-        var userTransactionStateVersionAndSuccessByPayloadHash = new Dictionary<string, CommittedTransactionKeyDetails>(committedTransactions.Count);
-
-        foreach (var committedTransaction in committedTransactions.Where(ct => ct.LedgerTransaction is CoreModel.UserLedgerTransaction))
-        {
-            var payloadHash = ((CoreModel.UserLedgerTransaction)committedTransaction.LedgerTransaction).NotarizedTransaction.HashBech32m;
-
-            var stateVersion = committedTransaction.ResultantStateIdentifiers.StateVersion;
-            var isSuccess = committedTransaction.Receipt.Status switch
-            {
-                CoreModel.TransactionStatus.Succeeded => true,
-                CoreModel.TransactionStatus.Failed => false,
-                _ => throw new UnreachableException($"Didn't expect {committedTransaction.Receipt.Status} value"),
-            };
-
-            var errorMessage = committedTransaction.Receipt.ErrorMessage;
-
-            userTransactionStateVersionAndSuccessByPayloadHash[payloadHash] = new CommittedTransactionKeyDetails(stateVersion, isSuccess, errorMessage);
-        }
+        var payloadHashes = committedTransactions
+            .Where(ct => ct.LedgerTransaction is CoreModel.UserLedgerTransaction)
+            .Select(ct => ((CoreModel.UserLedgerTransaction)ct.LedgerTransaction).NotarizedTransaction.HashBech32m)
+            .ToList();
 
-        var payloadHashes = userTransactionStateVersionAndSuccessByPayloadHash.Keys.ToList();
-
-        var toUpdate = await dbContext
+        var pendingTransactions = await dbContext
             .PendingTransactions
-            .Where(pt =>
-                pt.LedgerDetails.PayloadLedgerStatus != PendingTransactionPayloadLedgerStatus.CommittedSuccess
-                && pt.LedgerDetails.PayloadLedgerStatus != PendingTransactionPayloadLedgerStatus.CommittedFailure)
             .Where(pt => payloadHashes.Contains(pt.PayloadHash))
+            .Where(pt => pt.LedgerDetails.PayloadLedgerStatus == PendingTransactionPayloadLedgerStatus.PermanentlyRejected)
+            .Select(pt => new
+            {
+                pt.PayloadHash,
+                pt.LedgerDetails.PayloadLedgerStatus,
+                pt.GatewayHandling.FirstSubmittedToGatewayTimestamp,
+                pt.LedgerDetails.LatestRejectionTimestamp,
+                pt.LedgerDetails.LatestRejectionReason,
+            })
```

Comment:
As we update TXs with completely separate UPDATE query guess we can add `.NoTracking()`?